src/app/CodeRepo/lib/backends/github/index.js

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+const log = require("log");
+const rp = require("request-promise");
+const moment = require("moment");
+const { AsyncEventEmitter } = require("emitter");
+const GithubEventEmitter = require("./github_event_emitter");
+
+const GITHUB_API_BASE = "https://api.github.com";
+const GITHUB_BASE = "https://github.com";
+
+/*
+To get started with github:
+1. blalbabla
+2. blalalalala
+*/
+
+class GithubBackend extends AsyncEventEmitter {
+    constructor(id, backend, Repository, Revision) {
+        super();
+        this.id = id;
+        this.backend = backend;
+        this.locks = {};
+        this.Repository = Repository;
+        this.Revision = Revision;
+
+        this.githubEmitter = new GithubEventEmitter();
+    }
+
+    async start() {
+        // Start event monitoring towards github
+
+        try {
+            await this._startMonitorEventStream();
+        } catch (err) {
+            log.error("Failed to setup local web server: ", err);
```

Comment:
Perhaps a more expressive error message, "Failed to setup GitHub web hook target http server"?